packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,122 @@
+import { Transaction } from "bitcoinjs-lib-v5";
+import {
+  unsignedMultisigTransaction, 
+  multisigRequiredSigners,
+  multisigTotalSigners,
+  multisigAddressType
+} from "@caravan/bitcoin";
+import BigNumber from "bignumber.js";
+import { RBFOptions, UTXO, TransactionOutput, FeeRate } from "./types";
+import { isRBFSignaled, estimateVirtualSize } from "./utils";
+
+/**
+ * Prepare an RBF (Replace-By-Fee) transaction.
+ * This function doesn't broadcast the transaction, it just prepares it.
+ * @param options RBF options including the original transaction and new fee rate
+ * @returns A new transaction ready for signing and broadcasting
+ */
+export function prepareRBFTransaction(options: RBFOptions): Transaction {
+  const {
+    transaction,
+    newFeeRate,
+    utxos,
+    subtractFromOutput = false,
+    cancelTransaction = false,
+    destinationAddress,
+    network,
+  } = options;
+
+  if (!isRBFSignaled(transaction)) {
+    throw new Error("Original transaction is not signaling RBF");
+  }
+
+  // Calculate new fee
+  const newFee = calculateNewFee(transaction, newFeeRate);
+
+  // Calculate total input amount
+  const totalInput = utxos.reduce(
+    (sum, utxo) => sum.plus(utxo.value),
+    new BigNumber(0)
+  );
+
+  let outputs: TransactionOutput[] = [];
+
+  if (cancelTransaction) {
+    // For cancellation, send everything back to the destination address minus the new fee
+    const cancelAmount = totalInput.minus(newFee);
+    outputs.push({ address: destinationAddress!, amountSats: cancelAmount });
+  } else if (subtractFromOutput) {
+    // Subtract the fee increase from the first non-change output
+    const feeIncrease = newFee.minus(calculateFee(transaction, utxos));
+    outputs = transaction.outs.map((out, index) => {
+      if (index === 0 && !isChangeOutput(out, utxos)) {
+        return {
+          address: out.address!, // need to fix this implementation by defining a function to derive address from script
+          amountSats: new BigNumber(out.value).minus(feeIncrease),
+        };
+      }
+      return { address: out.address!, amountSats: new BigNumber(out.value) };
+    });
+  } else {
+    // Use the original outputs
+    outputs = transaction.outs.map((out) => ({
+      address: out.address!,
+      amountSats: new BigNumber(out.value),
+    }));
+  }
+
+  // Filter out dust outputs
+  outputs = outputs.filter((output) => output.amountSats.isGreaterThan(546));
+
+  // Use Caravan's unsignedMultisigTransaction function to create the new transaction
+  return unsignedMultisigTransaction(network, utxos, outputs, true); // true to enable RBF
```

Comment:
Yeah, sorry I should've mentioned this earlier but these are deprecated and should prefer using the psbt package. Feel free to add some deprecation notices that would've helped avoid this.

Reply:
You can look at [this PR](https://github.com/caravan-bitcoin/caravan/pull/91/files) where the new functions were introduced and how the legacy `unsignedMultisigTransaction` was swapped out for the new one. I tried to keep the api as similar as possible so it's not too hard to swap out.

Reply:
Thanks .... I'll add deprecation notices , wherever I find them ...